zanata-war/src/main/java/org/zanata/webtrans/server/HibernateIntegrator.java

```diff
@@ -37,6 +38,8 @@ public void integrate(Configuration configuration,
                     updateListener);
             eventListenerRegistry.appendListeners(EventType.POST_INSERT,
                     updateListener);
+            eventListenerRegistry.appendListeners(EventType.POST_COMMIT_UPDATE,
+                    ServiceLocator.instance().getInstance(SlugEntityUpdatedListener.class));
```

Comment:
It will be a single instance, but if it uses service locator in the scope of each method call, we guarantee that a different bean (because it's stateless) will be used every time the method is invoked, thus avoiding concurrency issues. Application scoped beans are very dangerous as we've come to learn.